### PR TITLE
HAX

### PR DESCRIPTION
**Pull request by clienthax**

### CODE REVIEW

File: src/main/java/com/cricketcraft/chisel/client/handler/DebugHandler.java [hunk 1]
```diff
@@ -58,7 +50,7 @@ public void onDrawScreen(RenderGameOverlayEvent.Post event){
         Minecraft mc = Minecraft.getMinecraft();
         IBakedModel m = mc.getBlockRendererDispatcher().getModelFromBlockState(state, mc.theWorld, mc.objectMouseOver.getBlockPos());
         if (m instanceof IFlexibleBakedModel.Wrapper){
-            m = (IBakedModel) ReflectionUtil.getValue("parent", m);
+//            m = (IBakedModel) ReflectionUtil.getValue("parent", m);
```
Comment (CptSpaceToaster) [inline src/main/java/com/cricketcraft/chisel/client/handler/DebugHandler.java:53]: This hack is pretty dirty... If `blockstate` is made public...can we clean up what's happening in the lines below.  (And remove the entirety of the if-statement)

File: src/main/java/com/cricketcraft/chisel/common/block/BlockCarvable.java [hunk 2]
```diff
@@ -124,12 +106,7 @@ public BlockCarvable(Material material, CarvableBlocks type, int subBlocksAmount
         this.index=index;
         this.VARIATION=p;
         this.fullBlock = isOpaqueCube();
-        if ((Boolean)Launch.blackboard.get("fml.deobfuscatedEnvironment")) {
-            ReflectionUtil.setFinalValue("blockState", Block.class, this, createRealBlockState(p));
-        }
-        else {
-            ReflectionUtil.setFinalValue("field_176227_L", Block.class, this, createRealBlockState(p));
-        }
+	    this.blockState = createRealBlockState(p);
```
Comment (CptSpaceToaster) [inline src/main/java/com/cricketcraft/chisel/common/block/BlockCarvable.java:109]: Indentation looks a little off

File: src/main/java/com/cricketcraft/chisel/common/block/BlockCarvable.java [hunk 3]
```diff
@@ -446,12 +423,7 @@ public static boolean blockStatesEqual(IBlockState state1, IBlockState state2){
                 return false;
             }
         }
-        if (state1.getBlock()==state2.getBlock()){
-            return true;
-        }
-        else {
-            return false;
-        }
+	    return state1.getBlock() == state2.getBlock();
```
Comment (CptSpaceToaster) [inline src/main/java/com/cricketcraft/chisel/common/block/BlockCarvable.java:426]: Indentation looks a little off

File: src/main/resources/mcmod.info [hunk 4]
```diff
@@ -1,16 +0,0 @@
...
-  "mcversion": "${mcversion}",
-  "url": "",
-  "updateUrl": "",
-  "authorList": ["ExampleDude"],
-  "credits": "The Forge and FML guys, for making this example",
-  "logoFile": "",
-  "screenshots": [],
-  "dependencies": []
-}
-]
```
Comment (CptSpaceToaster) [inline src/main/resources/mcmod.info:16]: Shouldn't we be tracking the mcmod.info, (with filled in information)?  I'm not certain it should be removed.

File: src/main/resources/META-INF/chisel_at.cfg [hunk 5]
```diff
@@ -0,0 +1 @@
+public-f net.minecraft.block.Block field_176227_L # blockState
```
Comment (CptSpaceToaster) [inline src/main/resources/META-INF/chisel_at.cfg:1]: It looks like you just removed the reflection helper and make the field that we were after public?

The diff is really noisy... but if you could confirm, that'd be great.

File: src/main/resources/mcmod.info (hunk 4 above)
Comment (clienthax) [inline src/main/resources/mcmod.info:16]: Considering it wasn't filled in, it was pointless to be in the repo at the time.

File: src/main/resources/META-INF/chisel_at.cfg (hunk 5 above)
Comment (clienthax) [inline src/main/resources/META-INF/chisel_at.cfg:1]: Yep, removed the reflection stuff for this field, as its easier to maintain the AT than using reflection for every block

**Pull request merged**